File: sdks/thunkmetrc-java/wrapper/src/main/java/io/github/thunkier/thunkmetrc/wrapper/models/Harvest.java
package io.github.thunkier.thunkmetrc.wrapper.models;import java.util.List;
import com.fasterxml.jackson.annotation.JsonProperty;public class Harvest {
    @JsonProperty("ArchivedDate")
    public String archivedDate;
    @JsonProperty("CurrentWeight")
    public Double currentWeight;
    @JsonProperty("DryingLocationId")
    public Integer dryingLocationId;
    @JsonProperty("DryingLocationName")
    public String dryingLocationName;
    @JsonProperty("DryingLocationTypeName")
    public String dryingLocationTypeName;
    @JsonProperty("DryingSublocationId")
    public Integer dryingSublocationId;
    @JsonProperty("DryingSublocationName")
    public String dryingSublocationName;
    @JsonProperty("FinishedDate")
    public String finishedDate;
    @JsonProperty("HarvestStartDate")
    public String harvestStartDate;
    @JsonProperty("HarvestType")
    public String harvestType;
    @JsonProperty("Id")
    public Integer id;
    @JsonProperty("IsOnHold")
    public Boolean isOnHold;
    @JsonProperty("IsOnInvestigation")
    public Boolean isOnInvestigation;
    @JsonProperty("IsOnInvestigationHold")
    public Boolean isOnInvestigationHold;
    @JsonProperty("IsOnInvestigationRecall")
    public Boolean isOnInvestigationRecall;
    @JsonProperty("LabTestingState")
    public String labTestingState;
    @JsonProperty("LabTestingStateDate")
    public String labTestingStateDate;
    @JsonProperty("LastModified")
    public String lastModified;
    @JsonProperty("Name")
    public String name;
    @JsonProperty("PackageCount")
    public Integer packageCount;
    @JsonProperty("PatientLicenseNumber")
    public String patientLicenseNumber;
    @JsonProperty("PlantCount")
    public Integer plantCount;
    @JsonProperty("SourceStrainCount")
    public Integer sourceStrainCount;
    @JsonProperty("SourceStrainNames")
    public String sourceStrainNames;
    @JsonProperty("Strains")
    public List<Object> strains;
    @JsonProperty("TotalPackagedWeight")
    public Double totalPackagedWeight;
    @JsonProperty("TotalRestoredWeight")
    public Double totalRestoredWeight;
    @JsonProperty("TotalWasteWeight")
    public Double totalWasteWeight;
    @JsonProperty("TotalWetWeight")
    public Double totalWetWeight;
    @JsonProperty("UnitOfWeightName")
    public String unitOfWeightName;
}

File: sdks/thunkmetrc-kotlin/wrapper/src/main/kotlin/io/github/thunkier/thunkmetrc/wrapper/models/Harvest.kt
package io.github.thunkier.thunkmetrc.wrapper.models

import com.fasterxml.jackson.annotation.JsonProperty



data class Harvest(
    @JsonProperty("ArchivedDate")
    val archivedDate: String? = null,
    @JsonProperty("CurrentWeight")
    val currentWeight: Double? = null,
    @JsonProperty("DryingLocationId")
    val dryingLocationId: Int? = null,
    @JsonProperty("DryingLocationName")
    val dryingLocationName: String? = null,
    @JsonProperty("DryingLocationTypeName")
    val dryingLocationTypeName: String? = null,
    @JsonProperty("DryingSublocationId")
    val dryingSublocationId: Int? = null,
    @JsonProperty("DryingSublocationName")
    val dryingSublocationName: String? = null,
    @JsonProperty("FinishedDate")
    val finishedDate: String? = null,
    @JsonProperty("HarvestStartDate")
    val harvestStartDate: String? = null,
    @JsonProperty("HarvestType")
    val harvestType: String? = null,
    @JsonProperty("Id")
    val id: Int? = null,
    @JsonProperty("IsOnHold")
    val isOnHold: Boolean? = null,
    @JsonProperty("IsOnInvestigation")
    val isOnInvestigation: Boolean? = null,
    @JsonProperty("IsOnInvestigationHold")
    val isOnInvestigationHold: Boolean? = null,
    @JsonProperty("IsOnInvestigationRecall")
    val isOnInvestigationRecall: Boolean? = null,
    @JsonProperty("LabTestingState")
    val labTestingState: String? = null,
    @JsonProperty("LabTestingStateDate")
    val labTestingStateDate: String? = null,
    @JsonProperty("LastModified")
    val lastModified: String? = null,
    @JsonProperty("Name")
    val name: String? = null,
    @JsonProperty("PackageCount")
    val packageCount: Int? = null,
    @JsonProperty("PatientLicenseNumber")
    val patientLicenseNumber: String? = null,
    @JsonProperty("PlantCount")
    val plantCount: Int? = null,
    @JsonProperty("SourceStrainCount")
    val sourceStrainCount: Int? = null,
    @JsonProperty("SourceStrainNames")
    val sourceStrainNames: String? = null,
    @JsonProperty("Strains")
    val strains: List<Any>? = null,
    @JsonProperty("TotalPackagedWeight")
    val totalPackagedWeight: Double? = null,
    @JsonProperty("TotalRestoredWeight")
    val totalRestoredWeight: Double? = null,
    @JsonProperty("TotalWasteWeight")
    val totalWasteWeight: Double? = null,
    @JsonProperty("TotalWetWeight")
    val totalWetWeight: Double? = null,
    @JsonProperty("UnitOfWeightName")
    val unitOfWeightName: String? = null
)
